source/common/network/proxy_protocol.cc

```diff
@@ -110,7 +110,7 @@ void ProxyProtocol::ActiveConnection::onReadWorker() {
 
   removeFromList(parent_.connections_);
 
-  listener.newConnection(fd, remote_address, local_address);
+  listener.newConnection(fd, remote_address, local_address, true);
```

Comment:
Semantically this is the same case as with SO_ORIGINAL_DST, as the local address comes from the proxy protocol header, and is the original destination of the connection, rather than from the listener's listening address. Therefore 'true' is right here.

This only has effect if the request is routed to an original destination cluster, so this is fully backwards compatible, and enables use of the original destination cluster also with traffic coming via an entity adding a proxy protocol header.